augur/merge.py

```diff
@@ -0,0 +1,304 @@
+"""
+Merge two or more metadata tables into one.
+
+Tables must be given unique names to identify them in the output and are
+merged in the order given.
+
+Rows are joined by id (e.g. "strain" or "name" or other
+--metadata-id-columns), and ids must be unique within an input table (i.e.
+tables cannot contain duplicate ids).  All rows are output, even if they
+appear in only a single table (i.e. a full outer join in SQL terms).
+
+Columns are combined by name, either extending the combined table with a new
+column or overwriting values in an existing column.  For columns appearing in
+more than one table, non-empty values on the right hand side overwrite values
+on the left hand side.  The first table's id column name is used as the output
+id column name.
+
+One generated column per input table is appended to the end of the output
+table to identify the source of each row's data.  Column names are generated
+as "__source_metadata_{NAME}" where "{NAME}" is the table name given to
+--metadata.  Values in each column are 1 or 0 for present or absent in that
+input table.
+
+Metadata tables of arbitrary size can be handled, limited only by available
+disk space.  Tables are not required to be entirely loadable into memory.  The
+transient disk space required is approximately the sum of the uncompressed size
+of the inputs.
+
+SQLite is used behind the scenes to implement the merge, but, at least for now,
+this should be considered an implementation detail that may change in the
+future.  The SQLite 3 CLI, sqlite3, must be available.  If it's not on PATH (or
```

Comment:
This means we should probably add sqlite3 CLI as a dependency for augur's bioconda recipe. I'll make an augur issue to ensure this is not forgotten (if we forget at first it's not too bad, we can fix the recipe any time).

Reply:
Made an issue: #1567

Reply:
Yep, I have this change uncommitted locally.

```diff
diff --git a/recipes/augur/meta.yaml b/recipes/augur/meta.yaml
index 582384a73f..68b26784d6 100644
--- a/recipes/augur/meta.yaml
+++ b/recipes/augur/meta.yaml
@@ -44,6 +44,7 @@ requirements:
     - mafft
     - raxml
     - vcftools
+    - sqlite >=3.39,<4
 
 test:
   imports:
```

Thanks for making an issue though!